internal/tiltd/tiltd_server/tiltd.go

```diff
@@ -5,19 +5,55 @@ import (
 	"os"
 	"os/exec"
 
+	"github.com/docker/docker/client"
+	"github.com/windmilleng/tilt/internal/build"
 	"github.com/windmilleng/tilt/internal/k8s"
 	"github.com/windmilleng/tilt/internal/tiltd"
 )
 
-type Daemon struct{}
+type Daemon struct {
+	b build.Builder
+}
 
 var _ tiltd.TiltD = &Daemon{}
 
 func NewDaemon() (*Daemon, error) {
-	return &Daemon{}, nil
+	opts := make([]func(*client.Client) error, 0)
+	opts = append(opts, client.FromEnv)
+
+	// Use client for docker 17
+	// https://docs.docker.com/develop/sdk/#api-version-matrix
+	// API version 1.30 is the first version where the full digest
+	// shows up in the API output of BuildImage
+	opts = append(opts, client.WithVersion("1.30"))
+	dcli, err := client.NewClientWithOpts(opts...)
+	if err != nil {
+		return nil, err
+	}
+	b := build.NewLocalDockerBuilder(dcli)
+	return &Daemon{b: b}, nil
 }
 
-func (d *Daemon) CreateService(ctx context.Context, k8sYaml string) error {
+func (d *Daemon) CreateService(ctx context.Context, k8sYaml string, dockerfile string, mounts []build.Mount, steps []build.Cmd, dockerfileTag string) error {
+	digest, err := d.b.BuildDocker(ctx, dockerfile, mounts, steps)
+	if err != nil {
+		return err
+	}
+	err = d.b.PushDocker(ctx, dockerfileTag, digest)
+	if err != nil {
+		return err
+	}
+
+	_, err = k8s.ParseYAMLFromString(k8sYaml)
+	if err != nil {
+		return err
+	}
+
+	// TODO(dmiller): not sure how to use this API
```

Comment:
@nicks how should I be using this?

Reply:
you need to iterate thru the entities and check that at least one has the image you want